src/components/LinkCardFavorite/LinkCardFavorite.tsx

```diff
@@ -0,0 +1,33 @@
+import { IconButton } from '@material-ui/core'
+import FavoriteIcon from '@material-ui/icons/Favorite'
+import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder'
+import firebase from 'firebase'
+import { useToggle } from 'react-use'
+
+import { Collections } from '../../enums/firebaseCollections'
+
+import type { LinkCardFavoriteProps } from './LinkCardFavorite.types'
+
+export const LinkCardFavorite: React.FunctionComponent<LinkCardFavoriteProps> = (props) => {
+    const { link } = props
+
+    const [isFavorite, toggleIsFavorite] = useToggle(false)
```

Comment:
You shouldn't have this here as you are duplicating the state.

If a user clicks on favorite really fast, it could get out of sync with the database